src/packagedcode/conan.py

```diff
@@ -0,0 +1,281 @@
+                self.requires.append(node.args[0].value)
+
+
+class ConanFileHandler(models.DatafileHandler):
+    datasource_id = "conan_conanfile_py"
+    path_patterns = ("*/conanfile.py",)
+    default_package_type = "conan"
+    default_primary_language = "C++"
+    description = "conan recipe"
+    documentation_url = "https://docs.conan.io/2.0/reference/conanfile.html"
+
+    @classmethod
+    def parse(cls, location):
+        with io.open(location, encoding="utf-8") as loc:
+            conan_recipe = loc.read()
+
+        try:
+            tree = ast.parse(conan_recipe)
+            recipe_class_def = next(
+                (
+                    node
+                    for node in tree.body
+                    if isinstance(node, ast.ClassDef)
+                    and node.bases
+                    and isinstance(node.bases[0], ast.Name)
+                    and node.bases[0].id == "ConanFile"
+                ),
+                None,
+            )
+
+            parser = ConanFileParser()
+            parser.visit(recipe_class_def)
+        except SyntaxError as e:
+            if TRACE:
+                logger_debug(f"Syntax error in conan recipe: {e}")
+            return
+
+        if TRACE:
+            logger_debug(f"ConanFileHandler: parse: package: {parser.to_dict()}")
+
+        dependencies = get_dependencies(parser.requires)
+
+        yield models.PackageData(
+            datasource_id=cls.datasource_id,
+            type=cls.default_package_type,
+            primary_language=cls.default_primary_language,
+            namespace=None,
+            name=parser.name,
+            version=parser.version,
+            description=parser.description,
+            homepage_url=parser.homepage_url,
+            keywords=parser.keywords,
+            extracted_license_statement=parser.license,
+            dependencies=dependencies,
+        )
+
+
+class ConanDataHandler(models.DatafileHandler):
+    datasource_id = "conan_conandata_yml"
+    path_patterns = ("*/conandata.yml",)
+    default_package_type = "conan"
+    default_primary_language = "C++"
+    description = "conan external source"
+    documentation_url = (
+        "https://docs.conan.io/2/tutorial/creating_packages/"
+        "handle_sources_in_packages.html#using-the-conandata-yml-file"
+    )
+
+    @classmethod
+    def parse(cls, location):
+        with io.open(location, encoding="utf-8") as loc:
+            conan_data = loc.read()
+
+        conan_data = saneyaml.load(conan_data)
+        sources = conan_data.get("sources", {})
+
+        for version, source in sources.items():
+            sha256 = source.get("sha256", None)
+            urls = []
+            source_urls = source.get("url")
+            if isinstance(source_urls, str):
+                urls.append(source_urls)
+            elif isinstance(source_urls, list):
+                urls.extend(source_urls)
+
+            for url in urls:
```

Comment:
Here all the URLs in one section is for one package, so we should yield one package only? Possibly with the other urls stored somewhere else? Extra data maybe? Two packages having the same sha256 value probably should not have two seperate entries. Even after combining the package data from conandata.yml and conanfile.py we have duplicate package instance entries with same purls, which should not be the case.

Another thing, but not too important probably, as I assume the conandata.yml files are not seen in isolation:
Also if it is clear and is a standard for conan package archive filenames having the name and version, we should probably also try to get the name out of it so we have valid purls? This would not have any impact on assembly as they would be merged allright if the purl is same, but would have purls also in the package data output.